### PR TITLE
Loot pools

### LINKED ISSUE #7: Add loot for defeating enemies
When enemies and monsters die, they should drop loot based on a percentage chance.

### To-Do
- Add loot to enemy object
- Loot should have a description of what they are
- Loot drop chance should be based on a percentage
- Should be added to a player's inventory in their object once an enemy is defeated

### PR DESCRIPTION
**Pull request by TerrenceJChan**
Code now gives a drop after defeating an enemy. We should consider how to format displaying the inventory sometime in the future. Closes #7

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -93,18 +109,29 @@ const chooseWeighted = (options) => {
...
+ */
+const addItem = (drop) => {
+  let result = inventory.find(loot => loot.mat === drop.mat);
+  if (result) {
+    result.quantity += 1;
+  } else {
+    inventory.push({ mat: drop.mat, quantity: 1 });
+  }
+};
+
```
Comment (TerrenceJChan) [inline src/index.js]: Hey @foxfriends, did I do the commenting correctly?

File: src/index.js [hunk 2]
```diff
@@ -49,7 +54,18 @@ let fightInfo = {
...
 
+let inventory = [];
+
 // Command that begins fight
+
+const checkInv = () => {
+  let message = '';
+  for (let i = 0; i < inventory.length; i++) {
+    message += inventory[i].mat + ' | ' + inventory[i].quantity + '\n';
+  }
```
Comment (foxfriends) [inline src/index.js:66]: 99% of the time you can use a `for..of` loop. Classic style indexed for loops like this are the way of the past (and much more error prone!).

On another note, it is usually preferred to use the backtick template strings when building strings out of other strings, as it's much more readable.


```suggestion
  for (const item of inventory) {
    message += `${item.mat} | ${item.quantity}\n`;
  }
```

File: src/index.js [hunk 3]
```diff
@@ -120,13 +147,20 @@ const attack = () => {
...
     switch (true) {
-    case enemy.hp <= 0:
-      return genericMsg + ' ' + enemy.msgs.defeat;
-    case enemy.hp <= fightInfo.startingEnemyHp * 0.5 && enemy.msgs.sub50[1] === false:
+    case enemy.hp <= 0: {
+      fightState = false;
+      const reward = chooseWeighted(enemy.drops);
+      addItem(reward);
+      console.log(inventory);
+      return (genericMsg + ' ' + enemy.msgs.defeat + '\n\n' + `You collect one <${reward.mat}> and place it in your inventory.`);
```
Comment (foxfriends) [inline src/index.js:156]: heh you funny, i never seen a hybrid template non template string before
```suggestion
      return `${genericMsg} ${enemy.msgs.defeat}\n\nYou collect one <${reward.mat}> and place it in your inventory.`;
```

**Review by foxfriends — APPROVED** [review]
Looks good! I might put object in lower case, as that has slightly different connotation than object in upper case, but not big deal either way

**Review by TerrenceJChan — COMMENTED** [review]
@foxfriends, can you have a look at my new `addItem()` function and see if the JSDocs formatting is done correctly?

File: src/index.js (hunk 1 above)
Comment (foxfriends) [reply src/index.js]: oho now i see it

Comment (foxfriends) [reply src/index.js]: anyway, only change i would suggest is that capital O in `Object`
```suggestion
/**
 * @param {object} drop An item to be added to the player's inventory
 * @param {string} drop.mat The item's name
 */
const addItem = (drop) => {
  let result = inventory.find(loot => loot.mat === drop.mat);
  if (result) {
    result.quantity += 1;
  } else {
    inventory.push({ mat: drop.mat, quantity: 1 });
  }
};

```

**Pull request merged**